File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        final int n = nums.length;
        final Set<Integer> set = new HashSet<Integer>();
        double expectedSum = n * (n + 1) * ((double) 1/2);
        int currSum = 0;
        for(int num : nums) {
            currSum += num;
        }
        int diff = (int) expectedSum - currSum;
        for(int num : nums) {
            if(set.contains(num)) {
                return new int[]{num,  num + diff};
            }
            set.add(num);
        }
        return new int[2];
    }
}

File: 0645-set-mismatch/0645-set-mismatch.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function(nums) {
    const n = nums.length;
    let expectedSum = n * ( n + 1 ) * (1 / 2);
    let sum = 0;
    for(let num of nums) {
        sum += num;
    }
    
    let diff = expectedSum - sum ;
    let set = new Set();
    for(let num of nums) {
        if(set.has(num)) {
            return [num, num + diff];
        }
        set.add(num);
    }
    return [];
};